packages/liveblocks-react/src/room.tsx

```diff
@@ -1475,56 +1505,82 @@ function useEditThreadMetadata<M extends BaseMetadata>() {
  * const createComment = useCreateComment();
  * createComment({ threadId: "th_xxx", body: {} });
  */
-function useCreateComment(): (options: CreateCommentOptions) => CommentData {
+function useCreateComment(): (options: CreateCommentOptions) => void {
   const client = useClient();
-  const room = useRoom();
+
+  // XXX - `useCreateComment` previously returned `CommentData`, but the signature is
+  // XXX - now updated to return `void` instead because there could be a no-op path
+  // XXX - if the specified thread could not be found. This signature is also consistent
+  // XXX - with `useEditComment` hook. Note: This would be a breaking change!!!
   return React.useCallback(
-    ({ threadId, body, attachments }: CreateCommentOptions): CommentData => {
+    ({ threadId, body, attachments }: CreateCommentOptions): void => {
       const commentId = createCommentId();
       const createdAt = new Date();
 
+      const userId = client[kInternal].currentUserIdStore.get();
+      // XXX - Should we display a console warning and return early if the user id could not be found?
+      if (userId === undefined) return;
```

Comment:
@marcbouchenoire @nvie What are your thoughts on how we should handle the lack of user id or thread in cache? 

This method currently (in `main`) returns the optimistically created `CommentData` object synchronously. But, if we do decide to return early (due to absence of user id or thread in cache), we'll be forced to update the signature of this method to return void. I think it is okay for this method to return a void object (the `useEditComment` hook returns a method that returns void too), but it would be a breaking change. Curious to hear your thoughts on this!

Reply:
Could we mimic what happens today with a public key scenario? What does the optimistic `CommentData` look like? What about the real server one? I don't think it's a warning/error case since we support it already via the public key today, even though it's not really useful.